department-of-veterans-affairs/abd-vro
# EP Merge: Added abort for ineligible claim requests

## What was the problem?

Currently, EP Merge does not abort jobs that should have been ineligible from the start. 

Associated tickets or Slack threads:
- #3241 

## How does this fix it?

Adds an ABORTING and ABORTED state. See acceptance criteria in the ticket for when jobs should be aborted.

This is the old state machine:

![EP_process (1)](https://github.com/user-attachments/assets/1fa3801b-8290-4404-8a15-8eec02e30fe1)

And this is the new implementation:
![EP_process](https://github.com/user-attachments/assets/78a353d2-7e43-4849-a463-5a4ac621c7a1)


## How to test this PR
- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run `pytest`
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
  - from /domain-ee/ee-ep-merge-app run `pytest ./integration`
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - `export BIP_CLAIM_URL=mock-bip-claims-api:20300`
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES=“all” ./gradlew dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`
